### PR TITLE
Make updateKeyValueStream reject unsupported options

### PR DESCRIPTION
**Pull request by mvo5**
This also improves the tests and superseeds https://github.com/snapcore/snapd/pull/3746

### CODE REVIEW

File: corecfg/proxy.go [hunk 1]
```diff
@@ -29,9 +29,9 @@ import (
 )
 
 var proxyConfigKeys = map[string]bool{
-	"http":  true,
-	"https": true,
-	"ftp":   true,
+	"http_proxy":  true,
```
Comment (pedronis) [inline corecfg/proxy.go]: are these fixes?  should the description mention them?

File: corecfg/utils_test.go [hunk 2]
```diff
@@ -31,24 +31,35 @@ type utilsSuite struct{}
...
 
+func (s *utilsSuite) TestUpdateKeyValueStreamConfigNotInAllConfig(c *C) {
+	in := bytes.NewBufferString("")
+	newConfig := map[string]string{"unsupported-options": "cannot be set"}
+	supportedConfigKeys := map[string]bool{
+		"foo": true,
+	}
+
+	_, err := corecfg.UpdateKeyValueStream(in, supportedConfigKeys, newConfig)
+	c.Check(err, ErrorMatches, `cannot set unsupported configuration value \"unsupported-options"`)
```
Comment (pedronis) [inline corecfg/utils_test.go]: \ is wrong as zyga noted later

Comment (zyga) [inline corecfg/utils_test.go]: Fixed

File: corecfg/proxy.go (hunk 1 above)
Comment (zyga) [inline corecfg/proxy.go]: This looks OK upon code inspection. I second @pedronis' comment though. It should be a separate commit with a description of what is going on. It seems this was just a bug all along and we didn't notice because of a bug in the `updateKeyValueStream`. With that fix this one surfaced.

Comment (mvo5) [inline corecfg/proxy.go]: This is properly fixed and tested in https://github.com/snapcore/snapd/pull/3754 - maybe we should focus on this one instead than this one here? Unfortunately removing this change will break the tests (because the bugfix in here exposes a bug in the proxy handling)
